### PR TITLE
enable markdown linter

### PR DESCRIPTION
**Pull request by danmoseley**
Enable markdown linter.

Most of [the rules](https://github.com/DavidAnson/markdownlint/blob/main/README.md) IMO are low value but linting does catch some things worth catching such as broken links and anchors and fenced blocks without specified language.

Disabled rules that are flagging issues we don't care about like trailing whitespace and line length. Fixed a few trivia though, and one broken link.

### CODE REVIEW

File: .markdownlint.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    "no-trailing-punctuation":         false,
+    "ol-prefix":                       { "one_or_ordered": true },
+    "blanks-around-fences":            false,
+    "blanks-around-lists":             false,
+     "no-inline-html":                 { "allowed_elements": [ "summary", "details" ]},
+    "no-bare-urls":                    false,
+    "single-trailing-newline":         false,
+    "emphasis-style":                  false,
+
+    // rule settings and options are documented in https://github.com/DavidAnson/markdownlint
```
Comment (danmoseley) [inline .markdownlint.json:17]: I don't know how JSON.Parse works exactly but comments like this appear to be ignored so long as they're at the end

**Comment by danmoseley** [comment]
using intentional errors here I verified it works

File: docs/APIReviewProcess.md [hunk 2]
```diff
@@ -24,7 +26,7 @@ Before marking an issue as `api-ready-for-review`, make sure that the issue has
 * A short description that will help reviewers not familiar with this area.
 * The API changes in ref-assembly format. It's fine to link this to the generated ref-assembly-code in the PR. If the changes are to an area that does produce ref-assemblies, please write out what it would look like in ref-assembly format for us to review.
 
-```txt
+````text
```
Comment (adityamandaleeka) [inline docs/APIReviewProcess.md:29]: four ticks? Surprised it's happy about that 😄

Comment (adityamandaleeka) [reply docs/APIReviewProcess.md:29]: I see this in a few other files too.

Comment (danmoseley) [reply docs/APIReviewProcess.md:29]: how curious

**Review by adityamandaleeka — APPROVED** [review]

File: docs/APIReviewProcess.md (hunk 2 above)
Comment (danmoseley) [reply docs/APIReviewProcess.md:29]: turns out that it's intentional
https://github.github.com/gfm/#fenced-code-blocks
similar to https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/strings/#raw-string-literals
it's so you can include stuff that itself has 3 ticks.

I'll fix though.

Comment (danmoseley) [reply docs/APIReviewProcess.md:29]: hmm, but it should have detected they weren't balanced...

Comment (danmoseley) [reply docs/APIReviewProcess.md]: or more strictly, that they weren't balanced by the end of the file.

fixed

Comment (mitchdenny) [reply docs/APIReviewProcess.md]: Yeah I change between one tick and three ticks all the time when doing comments on issues/PRs. So its definitely a standard markdown thing.

**Pull request merged**